src/jit/codegencommon.cpp

```diff
@@ -2465,6 +2465,21 @@ bool CodeGen::genCreateAddrMode(GenTreePtr  addr,
         return false;
     }
 
+#if defined(_TARGET_ARM_) && !defined(LEGACY_BACKEND)
+    if (mode == -1)
+    {
+        if (rv1 != nullptr && rv2 == nullptr && cns == 0
+#if SCALED_ADDR_MODES
+            && mul == 0
+#endif
+            )
+        {
+            // we can just remove [rv1 + 0]
+            return false;
+        }
+    }
+#endif // defined(_TARGET_ARM_) && !defined(LEGACY_BACKEND)
+
```

Comment:
I can see why one might want to be conservative about this, but why wouldn't this change be applicable to all targets?